File: src/leetcode/2026/01/Solution434.java
public class Solution434 {
    public int countSegments(String s) {
        int count = 0;
        boolean inSegment = false;

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) != ' ') {
                if (!inSegment) {
                    count++;
                    inSegment = true;
                }
            } else {
                inSegment = false;
            }
        }

        return count;
    }

    public static void main(String[] args) {
        Solution434 solution = new Solution434();

        String test1 = "Hello, my name is John";
        System.out.println("Number of segments in \"" + test1 + "\": " + solution.countSegments(test1));

        String test2 = "   Leading and trailing spaces   ";
        System.out.println("Number of segments in \"" + test2 + "\": " + solution.countSegments(test2));

        String test3 = "SingleWord";
        System.out.println("Number of segments in \"" + test3 + "\": " + solution.countSegments(test3));

        String test4 = "    ";
        System.out.println("Number of segments in \"" + test4 + "\": " + solution.countSegments(test4));

        String test5 = "";
        System.out.println("Number of segments in \"" + test5 + "\": " + solution.countSegments(test5));
    }
}

File: src/leetcode/2026/01/Solution434.cpp
#include <string>
#include <iostream>
using namespace std;

class Solution434 {
public:
    int countSegments(string s) {
        int count = 0;
        bool inSegment = false;

        for (char c : s) {
            if (c != ' ') {
                if (!inSegment) {
                    count++;
                    inSegment = true;
                }
            } else {
                inSegment = false;
            }
        }

        return count;
    }
};

int main() {
    Solution434 solution;

    string s1 = "Hello, my name is John";
    int result1 = solution.countSegments(s1);
    // Expected output: 5
    cout << "Number of segments: " << result1 << endl;

    string s2 = "   ";
    int result2 = solution.countSegments(s2);
    // Expected output: 0
    cout << "Number of segments: " << result2 << endl;

    string s3 = "SingleWord";
    int result3 = solution.countSegments(s3);
    // Expected output: 1
    cout << "Number of segments: " << result3 << endl;

    return 0;
}